### ISSUE TITLE
Throws on invalid dates

### ISSUE THREAD

**Issue by KishanBagaria** [kriszyp/msgpackr#48]
```
> require('msgpackr').encode(new Date('invalid'))
Uncaught:
RangeError: The number NaN cannot be converted to a BigInt because it is not an integer
    at BigInt (<anonymous>)
    at Packr.pack (.../node_modules/msgpackr/dist/node.cjs:1627:41)
    at pack (.../node_modules/msgpackr/dist/node.cjs:1355:34)
    at Object.Packr.pack.encode (.../node_modules/msgpackr/dist/node.cjs:1101:5)

> require('cbor-x').encode(new Date('invalid'))
<Buffer c1 fb 7f f8 00 00 00 00 00 00>

> require('@msgpack/msgpack').encode(new Date('invalid'))
Uint8Array(15) [
  199, 12, 255, 0, 0, 0,
    0,  0,   0, 0, 0, 0,
    0,  0,   0
]

> require('msgpack-lite').encode(new Date('invalid'))
<Buffer c7 09 0d cb 7f f8 00 00 00 00 00 00>
```

**Comment by kriszyp** [comment]
Are you requesting that it not throw? Based on the fact that there is no way to represent invalid dates in MessagePack, as far as I can tell, throwing an error seems like the correct behavior, I think?

(For the others, cbor-x is also doing the right thing, since CBOR _does_ has a fairly obvious way to encode an invalid date, since it accepts 64-bit floats that have a well-defined NaN representation. It looks like @msgpack/msgpack is encoding it as "1970-01-01 00:00:00 UTC" (seems wrong), and msgpack-lite's encoding isn't a using the MessagePack timestamp, as far as I can tell.)

**Comment by KishanBagaria** [comment]
Yep, I think it shouldn't throw by default and if it does the error message can be more descriptive rather than just mentioning BigInt. I almost shipped this into production without realizing. Switched to cbor-x just to have invalid dates working.

Maybe there can be a flag which enables this with an extension type? The readme should mention this behavior regardless. 

Not sure how you're implementing this but a sufficiently large BigInt will also be an invalid date:

```
> new Date(1e16)
Invalid Date
> BigInt(1e16)
10000000000000000n
> new Date(1e15)
+033658-09-27T01:46:40.000Z
```

**Comment by kriszyp** [comment]
Yes, a better error message certainly seems appropriate. I can certainly just come up with a invalid date representation. It probably would be reasonable to use the MessagePack timestamp code (-1), with zero length data to represent an invalid date. And would probably use a flag for this since it seems like most users prefer predictable fail-fast behavior. But this is a good suggestion, I will take a look at this.

**Comment by KishanBagaria** [comment]
Thanks for the quick fix!

**Comment by kriszyp** [comment]
Published in v1.5.1
